src/components/UserSelector.tsx

```diff
@@ -1,19 +1,60 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { GlobalContext } from './GeneralContext';
+import { User } from '../types/User';
+import { getPosts } from '../api/posts';
+import { ErrorType } from '../types/ErrorType';
 
 export const UserSelector: React.FC = () => {
+  const {
+    users,
+    selectedUser,
+    setSelectedUser,
+    setIsPostsLoading,
+    setPosts,
+    setError,
+    setSelectedPost,
+  } = useContext(GlobalContext);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const toggleDropdown = () => {
+    return isDropdownOpen
+      ? setIsDropdownOpen(false)
+      : setIsDropdownOpen(true);
+  };
+
+  const handleSetUser = (user: User) => {
+    setError(ErrorType.none);
+    setSelectedPost(null);
+    setIsPostsLoading(true);
+    setSelectedUser(user);
+    setIsDropdownOpen(false);
+    getPosts(user.id).then(setPosts)
+      .catch((err) => {
+        setError(ErrorType.postsLoadingError);
+        throw err;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)